recipe/install.py

```diff
@@ -0,0 +1,167 @@
+# This is free and unencumbered software released into the public domain.
```

Comment:
You have used this probably before/somewhere else? I'm a bit confused we need such a lengthy script. In most other cases, I either installed the shared libraries explicitly or had a script which deletes some files from the PREFIX again and then uses conda-build's host deduplication.

Reply:
> You have used this probably before/somewhere else? I'm a bit confused we need such a lengthy script.

Yes. The length is because I want to use this same script in multiple feedstocks without changing it too much, so it needs to be generic (long).

> In most other cases, I either installed the shared libraries explicitly or had a script which deletes some files from the PREFIX again and then uses conda-build's host deduplication.

Could you please explain better the approach you are describing here? I don't see how deleting files from the prefix is helpful for package splitting.

Reply:
See the script in https://github.com/conda-forge/glib-feedstock/blob/2896c82bbbe09c898d3d1bd9f5668bfb22571b00/recipe/install.sh#L1 There we install everything as usual and remove afterwards the files that should not go into the respective output.

Reply:
This is an interesting approach which we can call split-by-subtraction. It is the opposite of my approach which is split-by-addition. split-by-subtraction relies on a behavior of conda-build that the state of PREFIX is reset between generating each output (i.e. that the files which have been deleted are restored). The advantage is that hardcoded PREFIX paths in the installed artifacts do not need patching because they are installed directly to the PREFIX instead of to a staging area as in split-by-addition.

However, this restoration behavior between generating each output is undocumented, so relying on it as the primary package splitting method is (in my opinion) not a good idea because it could break at any time without warning. In fact, the only [documented methods](https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#specifying-files-to-include-in-output) are pattern matching via glob and moving files *into* the prefix (split-by-addition).

I have [an open PR](https://github.com/conda/conda-build/pull/5216) to improve the glob (pattern matching) functionality in conda-build which will make both split-by-subtraction and split-by-addition obsolete, so I can refactor this PR to use that feature if/when it is merged and released.

Reply:
> split-by-subtraction relies on a behavior of conda-build that the state of PREFIX is reset between generating each output (i.e. that the files which have been deleted are restored).

> However, this restoration behavior between generating each output is undocumented

The behaviour is clearly documented. Each output has its own set of `host` and `build` dependencies and thus gets its own environments during the build. We also never delete any files during a build phase that we have no copied into the PREFIX some lines above in the build script.